packages/node_modules/pouchdb-merge/src/removeLeafFromTree.js

```diff
@@ -0,0 +1,52 @@
+import { clone } from 'pouchdb-utils';
+
+// this method removes a leaf from a rev tree, independent of its status.
+// e.g., by removing an available leaf, it could leave its predecessor as
+// a missing leaf and corrupting the tree.
+function removeLeafFromRevTree(tree, leafRev) {
+  const _tree = clone(tree);
+  const toVisit = _tree.slice();
+
+  // we will store branchings (node.branches > 1) as a LIFO stack
+  const branchings = [];
```

Comment:
```suggestion
  const branchingNodes = [];
```
To distinct the purpose of this array a little more from the `branches` array.